gc/base/standard/ConcurrentGC.cpp

```diff
@@ -1122,21 +856,8 @@ MM_ConcurrentGC::conHelperEntryPoint(OMR_VMThread *omrThread, uintptr_t workerID
 
 		spinLimiter.reset();
 
-		/* clean cards */
-		while ((CONCURRENT_HELPER_MARK == request)
-				&& (CONCURRENT_CLEAN_TRACE == _stats.getExecutionMode())
-				&& _cardTable->isCardCleaningStarted()
-				&& !_cardTable->isCardCleaningComplete()
-				&& spinLimiter.spin()) {
-			if (cleanCards(env, false, _conHelperCleanSize, &sizeTraced, false)) {
-				if (sizeTraced > 0 ) {
-					_stats.incConHelperCardCleanCount(sizeTraced);
-					totalScanned += sizeTraced;
-					spinLimiter.reset();
-				}
-			}
-			request = getConHelperRequest(env);
-		}
+		/* Collector specific Con Helper Work */
+		conHelperDoWorkInternal(env, &request, &spinLimiter, &totalScanned);
```

Comment:
Should the very first `spinLimiter.reset();` be moved inside `MM_ConcurrentGCIncrementalUpdate::conHelperDoWorkInternal()` too?

Reply:
Thanks for pointing that out - yes, I think it should.